_scripts/run_notebook.py

```diff
@@ -0,0 +1,85 @@
+import json
+import subprocess
+
+import jupytext
+
+tsconfig_path = "tsconfig.json"
+
+
+def extract_ts_code(file_path: str) -> str:
+    with open(file_path, "r", encoding="utf-8") as notebook_file:
+        notebook_content = notebook_file.read()
+        notebook = jupytext.reads(notebook_content, fmt="ipynb")
+        ts_content = jupytext.writes(notebook, fmt="py:percent")
+
+    # Strip all lines starting with "#"
+    ts_content = "\n".join(
+        [line for line in ts_content.split("\n") if not line.startswith("#")]
+    )
+    return ts_content
+
+
+def get_tsconfig_options(tsconfig_path: str) -> list:
+    with open(tsconfig_path, "r", encoding="utf-8") as tsconfig_file:
+        tsconfig = json.load(tsconfig_file)
+        compiler_options = tsconfig.get("compilerOptions", {})
+        options = []
+        for key, value in compiler_options.items():
+            options.append(f"--{key}")
+            if isinstance(value, bool):
+                continue
+            elif isinstance(value, list):
+                options.append(",".join([str(item) for item in value]))
+            else:
+                options.append(str(value))
+    return options
+
+
+def compile_ts_code(ts_code: str, tsconfig_options: list) -> str:
+    print("Compiling TypeScript code...")
+    temp_file_path = "temp.ts"
+    with open(temp_file_path, "w") as f:
+        f.write(ts_code)
+    cmd = ["npx", "tsc", temp_file_path] + tsconfig_options
```

Comment:
Can't you just pass the config file to tsc?

Reply:
You would think. Maybe you can. Doesn't seem like it though
1. https://www.typescriptlang.org/docs/handbook/compiler-options.html
2. https://arc.net/l/quote/lvbifppw

>  When input files are specified on the command line, tsconfig.json files are ignored.